gltf-json/src/builders.rs

```diff
@@ -0,0 +1,1073 @@
+    pub fn material(mut self, m: Index<Material>) -> Self {
+        self.material = Some(m);
+        self
+    }
+
+    /// The type of primitives to render.
+    pub fn mode(mut self, m: Mode) -> Self {
+        self.mode = Valid(m);
+        self
+    }
+
+    /// An array of Morph Targets, each  Morph Target is a dictionary mapping
+    /// attributes (only `POSITION`, `NORMAL`, and `TANGENT` supported) to their
+    /// deviations in the Morph Target.
+    pub fn targets(mut self, t: Vec<MorphTarget>) -> Self {
+        self.targets = Some(t);
+        self
+    }
+
+    pub fn build(self) -> Primitive {
+        Primitive {
+            attributes: self.attributes,
+            extensions: self.extensions,
+            extras: self.extras,
+            indices: self.indices,
+            material: self.material,
+            mode: self.mode,
+            targets: self.targets,
+        }
+    }
+}
+
+/// A typed view into a buffer view.
+///
+/// This builder has both required and optional members. The required members
+/// are handled by struct initialization, while optional members can be
+/// specified after calling the method .with_options().
+///
+/// There are two dependently optional members, max and min.
+/// glTF 2.0 specification:
+///   POSITION accessor must have min and max properties defined.
+///
+/// .min and .max aren't validated here. Implement validation later as a pass
+/// over the entire gltf structure including buffers and use it for both import
+/// and export.
+///
+pub struct AccessorBuilder {
+    /// The parent buffer view this accessor reads from.
+    pub buffer_view: Index<View>,
+
+    /// The offset relative to the start of the parent `BufferView` in bytes.
+    ///
+    /// glTF 2.0 specification:
+    ///     The offset of an accessor into a bufferView (i.e.,
+    ///     accessor.byteOffset) and the offset of an accessor into a buffer
+    ///     (i.e., accessor.byteOffset + bufferView.byteOffset) must be a
+    ///     multiple of the size of the accessor's component type.
+    ///
+    pub byte_offset: usize,
+
+    /// The number of components within the buffer view - not to be confused
+    /// with the number of bytes in the buffer view.
+    pub count: usize,
+
+    /// The data type of components in the attribute.
+    pub component_type: GenericComponentType,
+
+    /// Specifies if the attribute is a scalar, vector, or matrix.
+    pub type_: Type,
+
+    /// Minimum value of each component in this attribute.
+    /// POSITION accessor must have min and max properties defined.
+    pub min: Option<serde_json::Value>,
+
+    /// Maximum value of each component in this attribute.
+    /// POSITION accessor must have min and max properties defined.
+    pub max: Option<serde_json::Value>,
+}
+
+impl AccessorBuilder {
+    /// Advance to the optional stage of this builder.
+    pub fn with_options(self) -> AccessorBuilderWithOptions {
```

Comment:
Is it possible to not have to call `with_options` like `PrimitiveBuilder`?

Reply:
Yes, the easiest way would be to drop compile time checks for required vs optional parameters and return a result from .build. Please see my reply https://github.com/gltf-rs/gltf/pull/193#issuecomment-421555650 for some follow-up questions.